app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -99,17 +99,46 @@ def get_stack_outputs(stack_id, outputs)
 
       def get_stack_resources(name, group)
         resources = @tds.list_deployment_operations(name, group)
-        # exclude empty resources and resources with an action
-        resources.reject! do |r|
-          r.try(:properties).try(:target_resource).try(:id).nil? ||
-            r.properties.target_resource.try(:action_name)
+        # relying on deployment operations to collect resources; but each resource may appear multiple times
+        # consolidate multiple appearances into only one
+        pos = -1
+        resources.reject! do |resource|
+          pos += 1
+          resource.properties.try(:target_resource).nil? || resource_already_collected?(resources, resource, pos)
         end
 
```

Comment:
They are not true duplicated entries in the array. Each entry has uniq trace id for example. But the target id is what we care about which can be duplicated.